utils/misc.ml

```diff
@@ -423,6 +423,17 @@ let find_first_mono =
 
 (* String operations *)
 
+let split_null_separated s =
+  let[@tail_mod_cons] rec discard_last_sep = function
+    | [] | [""] -> []
+    | x :: xs -> x :: discard_last_sep xs
+  in
+  discard_last_sep (String.split_on_char '\000' s)
+
+let concat_null_separated = function
+  | [] -> ""
+  | l -> String.concat "\000" l ^ "\000"
```

Comment:
The use of `^` means that we copy the output twice, which is unnecessary and could be noticeable if the output is large. You could use `String.concat "\000" (l @ [""])` instead which only copies the list spine.

Reply:
changed